frontend/src/app/modules/project/project-triggers/project-triggers.component.ts

```diff
@@ -1,46 +1,101 @@
+import { ProjectTriggerInfo } from '@shared/models/project/project-trigger/project-trigger-info';
 import { TriggerService } from '@core/services/trigger.service';
 import { ToastrNotificationsService } from '@core/services/toastr-notifications.service';
+import { ActivatedRoute } from '@angular/router';
 
 @Component({
   selector: 'app-project-triggers',
   templateUrl: './project-triggers.component.html',
   styleUrls: ['./project-triggers.component.sass']
 })
 export class ProjectTriggersComponent implements OnInit {
-
-  currentProject: Project = {} as Project;
+  projectId: number;
   branches: string [] = ['master', 'dev'];
   selectedBranch: string;
   runOnShedule = true;
-
+  triggers: ProjectTriggerInfo[] = [];
   constructor(
     private triggerService: TriggerService,
-    private toastrService: ToastrNotificationsService)
-    { }
-
+    private toastrService: ToastrNotificationsService,
+    private route: ActivatedRoute
+    )
+    {
+      route.parent.params.subscribe(
+        (params) => this.projectId = params.projectId);
+    }
   ngOnInit(): void {
+    this.getTriggers();
+  }
+
+  getTriggers() {
+    this.triggerService.getTriggersByProjectId(this.projectId).subscribe(
+      (data) => this.triggers = data,
+      (error) => this.toastrService.showError(error.message, error.name)
+    );
   }
 
   createTrigger(cron: string)  {
     if (this.selectedBranch) {
 
      const newTrigger: NewProjectTrigger = {
-       projectId: this.currentProject.id.toString(),
+       projectId: this.projectId,
        branchHash: this.selectedBranch,
        cronExpression: cron
       };
-
+     console.log(newTrigger.cronExpression);
      this.triggerService.createTrigger(newTrigger).subscribe(
-        () => this.toastrService.showSuccess('project created'),
+        (data) => {
+          console.log(data);
+
+          this.triggers.push(data);
+          this.toastrService.showSuccess('trigger created');
+        },
         (error) => this.toastrService.showError(error.message, error.name)
       );
     }
   }
+  updateTrigger(upTrigger: UpdateTriggerCron) {
+    const index = this.triggers.findIndex(t => t.id === upTrigger.id);
+    console.log(upTrigger.cronExpression);
+    const trigger: ProjectTrigger = {
+      id: this.triggers[index].id,
+      projectId: this.triggers[index].projectId,
+      branchHash: this.triggers[index].branchHash,
+      cronExpression: upTrigger.cronExpression
+    };
+
+    this.triggerService.updateTrigger(trigger).subscribe(
+      (data) => {
+        this.triggers.splice(index, 1, data);
+        this.toastrService.showSuccess('trigger updated');
+      },
+      (error) => this.toastrService.showError(error.message, error.name)
+    );
 
+  }
+  deleteTrigger(id: number) {
+    const index = this.triggers.findIndex(t => t.id === id);
+    this.triggerService.deleteTrigger(id).subscribe(
+      (data) => {
+        this.triggers.splice(index, 1);
```

Comment:
Let's filter out needed element and reassign triggers instead of mutating current array